owncloudData/src/main/java/com/owncloud/android/data/storage/LocalStorageProvider.kt

```diff
@@ -86,6 +89,52 @@ sealed class LocalStorageProvider(private val rootFolderName: String) {
     @SuppressLint("UsableSpace")
     fun getUsableSpace(): Long = getPrimaryStorageDirectory().usableSpace
 
+    /**
+     * Checks if there is user data which does not have a corresponding account in the Account manager.
+     */
+    private fun getDanglingAccountDirs(context: Context?): List<File> {
+        val am = AccountManager.get(context)
+        val rootFolder = File(getRootFolderPath())
+        val danglingDirs = mutableListOf<File>()
+        for (dir in rootFolder.listFiles()) {
+            var dir_is_ok = false
+            if (dir.name.equals("tmp"))  {
+                dir_is_ok = true
+            } else {
+                for (a in am.accounts) {
+                    if (dir.name.equals(getEncodedAccountName(a.name))) {
+                        dir_is_ok = true
+                    }
+                }
+            }
+            if (!dir_is_ok) {
+               danglingDirs.add(dir)
+            }
+        }
+        return danglingDirs
+    }
+
+    private fun recursiveDeleteDir(dir: File) {
```

Comment:
I think that there are some kotlin extensions to do this, so we don't need to do it by ourselfs

try `dir.deleteRecursively()`

Reply:
thx